src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,38 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+
 public class SalaryInfo {
+    private static final int DATE = 0;
+    private static final int NAME = 1;
+    private static final int HOURS = 2;
+    private static final int SALARY = 3;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder builder = new StringBuilder("Report for period ").append(dateFrom)
+                .append(" - ").append(dateTo).append(System.lineSeparator());
+        for (String name : names) {
+            int income = 0;
+            for (String dataInfo : data) {
+                String[] dataParts = dataInfo.split(" ");
+                if (formatData(dataParts[DATE]).isAfter(formatData(dateFrom).minusDays(1))
```

Comment:
we can do `formatData(dateFrom)` and `formatData(dateTo)` only once in the very beginning